daemon/api_test.go

```diff
@@ -3607,6 +3612,64 @@ func (s *apiSuite) TestIsTrue(c *check.C) {
 	}
 }
 
+var readyToBuyTests = []struct {
+	Input error
+	Test  func(c *check.C, rsp *resp)
+}{
+	{
+		// Success
+		Input: nil,
+		Test: func(c *check.C, rsp *resp) {
+			c.Check(rsp.Status, check.Equals, http.StatusOK)
```

Comment:
Instead of embedding duplicated logic here, just add another field to the table:
```
{
        input: nil,
        status: http.StatusOK,
        resptype: ResponseTypeSync,
        response: true,
}

Then, the comparisons can be done in the test body itself.